Crash when clicking on deck options

###### Reproduction Steps

1. Press and hold on a deck.
2. Click on "Options"

###### Expected Result

Should go to the options menu for that deck. I have a couple of decks that do not exhibit this behaviour and they are in the same options group.

###### Actual Result

The program crashes.

###### Debug info

AnkiDroid Version = 2.9.6
Android Version = 6.0.1
ACRA UUID = 2715152f-1201-4c3b-882e-8b2535d4be30

###### Research
*Enter an [ x ] character to confirm the points below:*

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

Got collection via email. Crashes on 2.10alpha71. Similar to, if not duplicate of #6068.
```
2020-04-29 10:27:09.082 16752-16752/com.ichi2.anki E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.ichi2.anki, PID: 16752
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.DeckOptions}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.preference.ListPreference.setEntries(java.lang.CharSequence[])' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3430)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3614)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:86)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199)
        at android.os.Handler.dispatchMessage(Handler.java:112)
        at android.os.Looper.loop(Looper.java:216)
        at android.app.ActivityThread.main(ActivityThread.java:7625)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.preference.ListPreference.setEntries(java.lang.CharSequence[])' on a null object reference
        at com.ichi2.anki.DeckOptions.buildLists(DeckOptions.java:789)
        at com.ichi2.anki.DeckOptions$DeckPreferenceHack$Editor.commit(DeckOptions.java:443)
        at com.ichi2.anki.DeckOptions$DeckPreferenceHack$Editor.apply(DeckOptions.java:499)
        at android.preference.PreferenceManager.setNoCommit(PreferenceManager.java:706)
        at android.preference.PreferenceManager.inflateFromResource(PreferenceManager.java:328)
        at android.preference.PreferenceActivity.addPreferencesFromResource(PreferenceActivity.java:1549)
        at com.ichi2.anki.DeckOptions.onCreate(DeckOptions.java:648)
        at android.app.Activity.performCreate(Activity.java:7458)
        at android.app.Activity.performCreate(Activity.java:7448)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1286)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3409)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3614) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:86) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199) 
        at android.os.Handler.dispatchMessage(Handler.java:112) 
        at android.os.Looper.loop(Looper.java:216) 
        at android.app.ActivityThread.main(ActivityThread.java:7625) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987) 
```

----

Cause:

This line returns `null`

https://github.com/ankidroid/Anki-Android/blob/d6c9f9415b4fc5c7d2ea73b17001e27e1ea8e8c7/AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.java#L779
Cause:

Swallowed exception: 

https://github.com/ankidroid/Anki-Android/blob/d6c9f9415b4fc5c7d2ea73b17001e27e1ea8e8c7/AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.java#L154-L156

`finish()` does not stop `onCreate()`

Exception:

```
com.ichi2.utils.JSONException: org.json.JSONException: Value false at timer of type java.lang.Boolean cannot be converted to int
```

JSON Config:
```
{"id":1568615369480,"mod":1588149381,"name":"Train 1 new","usn":4659,"maxTaken":60,"autoplay":true,"timer":false,"replayq":true,"new":{"bury":true,"delays":[1.0,10.0],"initialFactor":2500,"ints":[1,4,7],"order":1,"perDay":1,"separate":true},"rev":{"bury":true,"ease4":1.3,"ivlFct":1.0,"maxIvl":36500,"perDay":100,"hardFactor":1.2,"fuzz":0.05,"minSpace":1},"lapse":{"delays":[10.0],"leechAction":0,"leechFails":8,"minInt":1,"mult":0.0},"dyn":false}
```

This looks like an upstream config regression, confounded by a bug in our code.

--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.java
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.java
@@ -46,6 +46,7 @@ import com.ichi2.anki.services.ReminderService;
 import com.ichi2.async.CollectionTask;
 import com.ichi2.libanki.Collection;
 import com.ichi2.libanki.Consts;
+import com.ichi2.libanki.DeckConfig;
 import com.ichi2.preferences.StepsPreference;
 import com.ichi2.preferences.TimePreference;
 import com.ichi2.themes.StyledProgressDialog;
@@ -106,7 +107,7 @@ public class DeckOptions extends AppCompatPreferenceActivity implements OnShared
                 mValues.put("deckConf", mDeck.getString("conf"));
                 // general
                 mValues.put("maxAnswerTime", mOptions.getString("maxTaken"));
-                mValues.put("showAnswerTimer", Boolean.toString(mOptions.getInt("timer") == 1));
+                mValues.put("showAnswerTimer", Boolean.toString(parseTimerValue(mOptions)));
                 mValues.put("autoPlayAudio", Boolean.toString(mOptions.getBoolean("autoplay")));
                 mValues.put("replayQuestion", Boolean.toString(mOptions.optBoolean("replayq", true)));
                 // new
@@ -155,6 +156,12 @@ public class DeckOptions extends AppCompatPreferenceActivity implements OnShared
                 finish();
             }
         }
+
+
+        private boolean parseTimerValue(JSONObject options) {
+            return DeckConfig.parseTimerOpt(options, true);
+        }
+
 
         public class Editor implements SharedPreferences.Editor {
 

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/Card.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/Card.java
@@ -336,11 +336,6 @@ public class Card implements Cloneable {
     }
 
 
-    public boolean shouldShowTimer() {
-        return mCol.getDecks().confForDid(mODid == 0 ? mDid : mODid).getInt("timer") != 0;
-    }
-
-
     /*
      * Time taken to answer card, in integer MS.
      */
@@ -588,7 +583,8 @@ public class Card implements Cloneable {
 
 
     public boolean showTimer() {
-        return mCol.getDecks().confForDid(mODid == 0 ? mDid : mODid).optInt("timer", 1) != 0;
+        JSONObject options = mCol.getDecks().confForDid(mODid == 0 ? mDid : mODid);
+        return DeckConfig.parseTimerOpt(options, true);
     }
 
 

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java
@@ -1,0 +1,46 @@
+/*
+ Copyright (c) 2020 David Allison <davidallisongithub@gmail.com>
+
+ This program is free software; you can redistribute it and/or modify it under
+ the terms of the GNU General Public License as published by the Free Software
+ Foundation; either version 3 of the License, or (at your option) any later
+ version.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along with
+ this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.ichi2.libanki;
+
+import com.ichi2.utils.JSONObject;
+
+import androidx.annotation.Nullable;
+
+public class DeckConfig {
+
+    public static @Nullable Boolean parseTimer(JSONObject config) {
+        //Note: Card.py used != 0, DeckOptions used == 1
+        try {
+            //#6089 - Anki 2.1.24 changed this to a bool, reverted in 2.1.25.
+            return config.getInt("timer") != 0;
+        } catch (Exception e) {
+            try {
+                return config.getBoolean("timer");
+            } catch (Exception ex) {
+                return null;
+            }
+        }
+    }
+
+    public static boolean parseTimerOpt(JSONObject config, boolean defaultValue) {
+        Boolean ret = parseTimer(config);
+        if (ret == null) {
+            ret = defaultValue;
+        }
+        return ret;
+    }
+}

